micro_manager/config.py

```diff
@@ -64,122 +64,166 @@ def read_json(self, config_filename):
-            assert isinstance(self._write_data_names, dict), "Write data entry is not a dictionary"
+            assert isinstance(
+                self._write_data_names, dict
+            ), "Write data entry is not a dictionary"
             for key, value in self._write_data_names.items():
                 if value == "scalar":
                     self._write_data_names[key] = False
                 elif value == "vector":
                     self._write_data_names[key] = True
                 else:
-                    raise Exception("Write data dictionary as a value other than 'scalar' or 'vector'")
+                    raise Exception(
+                        "Write data dictionary as a value other than 'scalar' or 'vector'"
+                    )
         except BaseException:
-            self._logger.info("No write data names provided. Micro manager will only read data from preCICE.")
+            self._logger.info(
+                "No write data names provided. Micro manager will only read data from preCICE."
+            )
 
         try:
             self._read_data_names = data["coupling_params"]["read_data_names"]
-            assert isinstance(self._read_data_names, dict), "Read data entry is not a dictionary"
+            assert isinstance(
+                self._read_data_names, dict
+            ), "Read data entry is not a dictionary"
             for key, value in self._read_data_names.items():
                 if value == "scalar":
                     self._read_data_names[key] = False
                 elif value == "vector":
                     self._read_data_names[key] = True
                 else:
-                    raise Exception("Read data dictionary as a value other than 'scalar' or 'vector'")
+                    raise Exception(
+                        "Read data dictionary as a value other than 'scalar' or 'vector'"
+                    )
         except BaseException:
-            self._logger.info("No read data names provided. Micro manager will only write data to preCICE.")
+            self._logger.info(
+                "No read data names provided. Micro manager will only write data to preCICE."
+            )
 
         self._macro_domain_bounds = data["simulation_params"]["macro_domain_bounds"]
 
         try:
             self._ranks_per_axis = data["simulation_params"]["decomposition"]
         except BaseException:
             self._logger.info(
-                "Domain decomposition is not specified, so the Micro Manager will expect to be run in serial.")
+                "Domain decomposition is not specified, so the Micro Manager will expect to be run in serial."
+            )
 
         try:
-            if data["simulation_params"]["adaptivity"]:
+            if data["simulation_params"]["adaptivity"] == "True":
                 self._adaptivity = True
             else:
                 self._adaptivity = False
```

Comment:
Could also warn the case when the adaptivity parameters are given but adaptivity is set to false.